client/my-sites/site-settings/form-seo.jsx

```diff
@@ -0,0 +1,279 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import get from 'lodash/get';
+import omitBy from 'lodash/omitBy';
+import isBoolean from 'lodash/isBoolean';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import Button from 'components/button';
+import SectionHeader from 'components/section-header';
+import ExternalLink from 'components/external-link';
+import Notice from 'components/notice';
+import NoticeAction from 'components/notice/notice-action';
+import notices from 'notices';
+import protectForm from 'lib/mixins/protect-form';
+import FormInput from 'components/forms/form-text-input-with-affixes';
+import FormInputValidation from 'components/forms/form-input-validation';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import CountedTextarea from 'components/forms/counted-textarea';
+import analytics from 'lib/analytics';
+import PageViewTracker from 'lib/analytics/page-view-tracker';
+
+export default React.createClass( {
+	displayName: 'SiteSettingsFormSEO',
+
+	mixins: [ protectForm.mixin ],
+
+	getInitialState() {
+		return this.stateForSite( this.props.site );
+	},
+
+	componentWillReceiveProps: function( nextProps ) {
+		if ( get( nextProps, 'site.ID', 0 ) !== get( this.props, 'site.ID', 0 ) ) {
+			if ( get( nextProps, 'site.jetpack' ) ) {
+				// Go back to general settings if switched to a Jetpack site
+				page( this.getGeneralSettingsUrl( get( nextProps, 'site.slug', '' ) ) );
+				return;
+			}
+
+			// Update state when switching sites
+			this.setState( this.stateForSite( nextProps.site ) );
+		}
+	},
+
+	stateForSite: function( site ) {
+		return {
+			seoMetaDescription: get( site, 'options.seo_meta_description', '' ),
+			verificationServicesCodes: {
+				google: get( site, 'options.verification_services_codes.google', '' ),
+				bing: get( site, 'options.verification_services_codes.bing', '' ),
+				pinterest: get( site, 'options.verification_services_codes.pinterest', '' ),
+				yandex: get( site, 'options.verification_services_codes.yandex', '' )
+			}
+		};
+	},
+
+	handleMetaChange( event ) {
+		this.setState( { seoMetaDescription: get( event, 'target.value', '' ) } );
+	},
+
+	handleVerificationCodeChange( event, serviceName ) {
+		const servicesCodes = this.state.verificationServicesCodes;
+		if ( ! servicesCodes.hasOwnProperty( serviceName ) ) {
+			return;
+		}
```

Comment:
is this a safety check? it's good code, and a good safety, but I also feel like it might be more clear that we want to prevent hacked service names by checking for inclusion in a list of valid names…

```js
import includes from 'lodash/includes';

// via an explicit list
if ( ! includes( [ 'google', 'bing', 'pinterest', 'yonder' ], serviceName ) ) {
	return;
}

// via keys…
if ( ! includes( Object.keys( serviceCodes ), serviceName ) ) {
	return;
}
```